src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.js

```diff
@@ -0,0 +1,237 @@
+import React, {
+    useContext, useEffect, useMemo,
+} from 'react';
+import {Dimensions, Keyboard} from 'react-native';
+import Reanimated, {
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming,
+} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import styles from '../../styles/styles';
+import {Actions, States, ActionSheetAwareScrollViewContext} from './ActionSheetAwareScrollViewContext';
+
+const config = {
+    duration: 350,
+    easing: Easing.bezier(0.33, 0.01, 0, 1),
+};
+
+function ActionSheetKeyboardSpace(props) {
+    const safeArea = useSafeAreaInsets();
+    const keyboard = useAnimatedKeyboard();
+
+    // similar to using `global` in worklet but it's just a local object
+    const ctx = useMemo(() => ({}), []);
+    const windowHeight = Dimensions.get('screen').height;
+    const {
+        currentActionSheetState, transitionActionSheetStateWorklet: transition, transitionActionSheetState, resetStateMachine,
+    } = useContext(ActionSheetAwareScrollViewContext);
+
+    useEffect(() => {
+        // sometimes it might trigger multiple times the same actions
+        let lastAction = null;
+        const onDidShow = () => {
+            if (lastAction === 'keyboardDidShow') {
+                return;
+            }
+
+            lastAction = 'keyboardDidShow';
+
+            transitionActionSheetState({
+                type: Actions.ON_KEYBOARD_OPEN,
+            });
+        };
+        const removeDidShow = Keyboard.addListener('keyboardDidShow', onDidShow);
+
+        const onDidHide = () => {
+            if (lastAction === 'keyboardDidHide') {
+                return;
+            }
+
+            lastAction = 'keyboardDidHide';
+
+            transitionActionSheetState({
+                type: Actions.ON_KEYBOARD_CLOSE,
+            });
+        };
+        const removeDidHide = Keyboard.addListener('keyboardDidHide', onDidHide);
+
+        return () => {
+            resetStateMachine();
+
+            // We try to use the new API first, but it doesn't work all the time
+            // Especially it's bad with hot reloading, so we fallback to the old API
+            try {
+                removeDidShow();
+                removeDidHide();
+            } catch (err) {
+                Keyboard.removeSubscription(onDidShow);
+                Keyboard.removeSubscription(onDidHide);
+            }
+        };
+    }, []);
+
+    // We need this because of the bug in useAnimatedKeyboard.
```

Comment:
Cool, again let's get the fix merged upstream (if it's not already merged), update Reanimated in a separate PR, then remove the workaround here.

Reply:
Cool, well even if it takes extra time I'd rather wait for the upstream fix v.s: implement a workaround

Reply:
@tomekzaw I created an issue in the Reanimated repo that shows the problem I have. I tried with both 3.0.2 and the latest commit as of today:
https://github.com/software-mansion/react-native-reanimated/issues/4324

TLDR: the keyboard height is updating and the `useAnimatedStyle` hook is triggered for each value change, but the styles won't update until I click somewhere on the screen.

Reply:
This seems to have been solved upstream, resolving this thread

Reply:
Actually, we need to remove the workaround from here in this PR